lightly/active_learning/scorers/classification.py

```diff
@@ -64,25 +92,73 @@ class ScorerClassification(Scorer):
         >>> scorer = ScorerClassification(predictions)
 
     """
-    def __init__(self, model_output: np.ndarray):
+    def __init__(self, model_output: Union[np.ndarray, List[List[float]]]):
+        if isinstance(model_output, List):
+            model_output = np.array(model_output)
         super(ScorerClassification, self).__init__(model_output)
 
-    def calculate_scores(self) -> Dict[str, np.ndarray]:
+    @classmethod
+    def score_names(cls) -> List[str]:
+        """Returns the names of the calculated active learning scores
+        """
+        score_names = list(cls(model_output=[[0.5, 0.5]]).calculate_scores().keys())
+        return score_names
+
+    def calculate_scores(self, normalize_to_0_1: bool = True) -> Dict[str, np.ndarray]:
         """Calculates and returns the active learning scores.
 
+        Args:
+            normalize_to_0_1:
+                If this is true, each score is normalized to have a
+                theoretical minimum of 0 and a theoretical maximum of 1.
+
         Returns:
             A dictionary mapping from the score name (as string)
             to the scores (as a single-dimensional numpy array).
         """
+        if len(self.model_output) == 0:
+            return {score_name: [] for score_name in self.score_names()}
+
+        scores_with_names = [
+            self._get_scores_uncertainty_least_confidence(),
+            self._get_scores_uncertainty_margin(),
+            self._get_scores_uncertainty_entropy()
+        ]
+
         scores = dict()
-        scores["prediction-margin"] = self._get_prediction_margin_score()
-        scores["prediction-entropy"] = self._get_prediction_entropy_score()
+        for score, score_name in scores_with_names:
+            score = np.nan_to_num(score)
+            scores[score_name] = score
+
+        if normalize_to_0_1:
+            scores = self.normalize_scores_0_1(scores)
+
+        return scores
+
+    def normalize_scores_0_1(self, scores: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
```

Comment:
I would like to investigate an option to only rescaling the max. E.g. making sure that max value is 1.0. But the min value could be 0.0 or 0.2 etc. 

I would make another (exploratory) issue for this. Do you think that makes sense @MalteEbner?

Reply:
The advantage of rescaling to the theoretical max instead of the max of the current samples is that it makes the scores comparable across different iterations.
Thus I think that there is no clearly better solution for this topic. Feel free to create an issue for outlining and discussing the different options. However, I feel that this is rather low-priority.